### PR TITLE
Fixes error with fit() with validation data

### LINKED ISSUE #364: [BUG] Model.fit() with validation_data arg is failing after the Retrieval models evaluation refactory
### Bug description
After this #352 the following error occurs.
When I run `model.fit()` setting `validation_data` argument, it will call `model.evaluate()` in the end of each epoch. Then `RetrievalModel.compute_loss_metrics()` is called and sets [here](https://github.com/NVIDIA-Merlin/models/blob/tf/retrieval-eval/merlin/models/tf/models/base.py#L596) `kwargs["eval_sampling"] = True`, so that in the `ItemRetrievalScorer.call()` the `eval_sampling` is `True`.
But the eval_sampling is not being passed through for some reason, which causes an error in `ItemRetrievalScorer` when using model.fit(..., validation_data=...)

### PR DESCRIPTION
**Pull request by gabrielspmoreira**
- Closes #364 
- Changes `filter_kwargs()` to support forwarding kwargs for functions that have **kwargs

### CODE REVIEW

File: merlin/models/tf/blocks/core/combinators.py [hunk 1]
```diff
@@ -209,19 +209,19 @@ def call(self, inputs, training=False, **kwargs):
...
-                        {**dict(training=training), **kwargs},
-                        layer.call,
-                        filter_positional_or_keyword=False,
-                    )
-                )
-            else:
-                filtered_kwargs = filter_kwargs(
-                    dict(training=training), layer, filter_positional_or_keyword=False
+            filtered_kwargs = filter_kwargs(
+                {**dict(training=training), **kwargs},
```
Comment (marcromeyn) [inline merlin/models/tf/blocks/core/combinators.py:213]:
```suggestion
                {training=training, **kwargs},
```

File: merlin/models/tf/blocks/core/combinators.py [hunk 2]
```diff
@@ -209,19 +209,19 @@ def call(self, inputs, training=False, **kwargs):
...
+                layer,
+                cascade_kwargs_if_possible=True,
+            )
+            filtered_kwargs.update(
+                filter_kwargs(
+                    {**dict(training=training), **kwargs},
```
Comment (marcromeyn) [inline merlin/models/tf/blocks/core/combinators.py:219]:
```suggestion
                    {training=training, **kwargs},
```

**Review by marcromeyn — APPROVED** [review]
Looks good, just left some small comments

File: merlin/models/tf/blocks/core/combinators.py (hunk 2 above)
Comment (gabrielspmoreira) [reply merlin/models/tf/blocks/core/combinators.py]: Thanks, just added your suggestion in my latest commit

**Pull request merged**